runtime/autoload/lsp/request.vim

```diff
@@ -0,0 +1,21 @@
+let g:nvim_lsp_client = -1
+
+function! s:get_client() abort
+   if g:nvim_lsp_client == -1
+      lua << EOF
+local client = require('runtime.lua.lsp.client')
```

Comment:
Okay, I have removed many of them and rebased, so I can use the updates from #6789 . I will clean up the code as I go along. Still brainstorming a lot of ideas.